client/dashboard/domains/name-servers/form.tsx

```diff
@@ -0,0 +1,219 @@
+// eslint-disable-next-line no-restricted-imports
+import { CHANGE_NAME_SERVERS_FINDING_OUT_NEW_NS } from '@automattic/urls';
+import {
+	Button,
+	ToggleControl,
+	__experimentalText as Text,
+	__experimentalView as View,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { createInterpolateElement } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { useCallback, useMemo, useState } from 'react';
+import Notice from '../../components/notice';
+import { getServiceName } from '../../utils/service-name';
+import { NameServerInput, validateField } from './form-input';
+import {
+	NameServerField,
+	MIN_NAMESERVER_LENGTH,
+	MAX_NAMESERVER_LENGTH,
+	WPCOM_DEFAULT_NAMESERVERS,
+} from './types';
+import UpsellNudge from './upsell-nudge';
+import { areAllWpcomNameServers } from './utils';
+import type { InstanceType } from '../../app/context';
+
+interface Props {
+	domainName: string;
+	instanceType: InstanceType;
+	showUpsellNudge?: boolean;
+	nameservers?: string[];
+	isBusy?: boolean;
+	queryError?: string;
+	onSubmit: ( nameServers: string[] ) => void;
+}
+
+export default function NameServersForm( {
```

Comment:
I really can't help but think we could benefit from a form library here. `react-hook-form` seem like the perfect abstraction for us.

Reply:
Could we use `DataForm` with basic validation? p1754433905422279-slack-C08JZTRS6NA

I'd refrain from coming up with another different form handling in the dashboard code. We've been using DataForm virtually for every form in the new dashboard. 🤔